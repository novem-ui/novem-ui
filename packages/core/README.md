# @novem-ui/core

This lib exports all the Novem UI components