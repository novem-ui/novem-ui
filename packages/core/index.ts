export { default as NovemProvider } from './src/novem-provider'

export { default as Badge } from '@novem-ui/badge'
export * from '@novem-ui/badge'

export { default as Button, ButtonProps } from '@novem-ui/button'

export { default as Switch, SwitchProps } from '@novem-ui/switch'

export { Heading, Label, Paragraph, HeadingProps, LabelProps, ParagraphProps } from '@novem-ui/text'

export { Input, TextField, TextFieldProps } from '@novem-ui/input'

export { default as theme } from '@novem-ui/theme'
export type { Theme } from '@novem-ui/theme'
