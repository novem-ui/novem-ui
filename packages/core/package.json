{
  "name": "@novem-ui/core",
  "version": "0.0.1-alpha.1",
  "description": "Core components",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/core#readme",
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/core"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/core --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "dependencies": {
    "@novem-ui/badge": "^0.0.1-alpha.1",
    "@novem-ui/theme": "^0.0.1-alpha.1"
  },
  "devDependencies": {
    "@types/novem-theme": "^0.0.1-alpha.1",
    "@types/react": "^17.0.3"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.3.0",
    "polished": "^4.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
