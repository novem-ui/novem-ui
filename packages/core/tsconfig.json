{
  "extends": "../../tsconfig.json",
  "files": [],
  "include": ["**/*.ts", "**/*.tsx", "../../emotion.d.ts"],
  "exclude": ["node_modules/**", "dist/**"]
}
