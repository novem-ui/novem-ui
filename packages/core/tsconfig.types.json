{
  "extends": "../../tsconfig.json",
  "files": [],
  "compilerOptions": {
    "emitDeclarationOnly": true,
    "declaration": true,
    "declarationDir": "../../dist/",
    "declarationMap": true,
    "types": ["novem-theme"]
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules/**", "dist/**", "**/*.spec.tsx", "**/*.stories.tsx"]
}
