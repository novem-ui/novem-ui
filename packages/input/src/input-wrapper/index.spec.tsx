import React from 'react'

import { render } from '@testing-library/react'
import { theme } from '@novem-ui/core'
import { matchers } from '@emotion/jest'

import InputWrapper, { InputWrapperProps } from '.'

beforeAll(() => {
  expect.extend(matchers)
})

const setup = (props: InputWrapperProps = {}) => {
  const { container } = render(<InputWrapper theme={theme} {...props} />)

  return { container }
}

describe('input/input', () => {
  test('should render an input wrapper with a valid state', () => {
    const { container } = setup({ valid: true })

    expect(container).toMatchSnapshot()
  })

  test('should render an input wrapper with an error state', () => {
    const { container } = setup({ error: true })

    expect(container).toMatchSnapshot()
  })

  test('should render an input wrapper with an disabled state', () => {
    const { container } = setup({ disabled: true })

    expect(container).toMatchSnapshot()
  })
})
