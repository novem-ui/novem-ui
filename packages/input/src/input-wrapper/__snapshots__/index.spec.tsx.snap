// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input/input should render an input wrapper with a valid state 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.emotion-0::before,
.emotion-0::after {
  content: '';
  display: block;
  top: 50%;
  position: absolute;
  right: 1rem;
}

.emotion-0::before {
  background-image: url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M16 0H0V16H16V0Z' fill='white' fill-opacity='0.01'/%3E%3Cpath d='M3.33301 7.99996L6.66634 11.3333L13.333 4.66663' stroke='white' stroke-width='1.33333' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
  height: 1rem;
  -webkit-transform: translate(-14%, -50%);
  -moz-transform: translate(-14%, -50%);
  -ms-transform: translate(-14%, -50%);
  transform: translate(-14%, -50%);
  width: 1rem;
  z-index: 1;
}

.emotion-0::after {
  background-color: #59BC6F;
  border-radius: 0.625rem;
  height: 1.25rem;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 1.25rem;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`input/input should render an input wrapper with an disabled state 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

<div>
  <div
    class="emotion-0"
    disabled=""
  />
</div>
`;

exports[`input/input should render an input wrapper with an error state 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.emotion-0::before,
.emotion-0::after {
  content: '';
  display: block;
  top: 50%;
  position: absolute;
  right: 1rem;
}

.emotion-0::before {
  background-image: url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M16 0H0V16H16V0Z' fill='white' fill-opacity='0.01'/%3E%3Cpath d='M4.66699 4.66663L11.3337 11.3333' stroke='white' stroke-width='1.33333' stroke-linecap='round' stroke-linejoin='round'/%3E%3Cpath d='M4.66699 11.3333L11.3337 4.66663' stroke='white' stroke-width='1.33333' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
  height: 1rem;
  -webkit-transform: translate(-14%, -50%);
  -moz-transform: translate(-14%, -50%);
  -ms-transform: translate(-14%, -50%);
  transform: translate(-14%, -50%);
  width: 1rem;
  z-index: 1;
}

.emotion-0::after {
  background-color: #EB5757;
  border-radius: 0.625rem;
  height: 1.25rem;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 1.25rem;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;
